src/TufValidatedComposerRepository.php

```diff
@@ -62,10 +58,12 @@ public function __construct(array $repoConfig, IOInterface $io, Config $config,
             // it doesn't already exist.
             $rootFile = $repoPath . '/root.json';
             if (!file_exists($rootFile)) {
-                $sourcePath = realpath($repoConfig['tuf']['root']);
-                if (!$fs->copy($sourcePath, $rootFile)) {
-                    throw new FilesystemException("Could not copy '$sourcePath' to '$rootFile");
-                }
+                $sourcePath = implode(DIRECTORY_SEPARATOR, [
+                    dirname($config->getConfigSource()->getName()),
+                    'tuf',
+                    $repoKey,
+                ]);
+                $fs->copy("$sourcePath.json", $rootFile);
             }
 
             // For unit testing purposes, allow the updater instance to be provided
```

Comment:
No change but just mention I am not sure this comment still makes sense

Reply:
Fixed.